core/src/main/scala/com/yahoo/maha/core/query/QueryPipeline.scala

```diff
@@ -837,8 +851,19 @@ class DefaultQueryPipelineFactory(implicit val queryGeneratorRegistry: QueryGene
           } else {
             if (factBestCandidateOption.isDefined) {
               val query = getDimFactQuery(bestDimCandidates, factBestCandidateOption.get, requestModel, queryAttributes)
+              val fallbackQueryOption: Option[(Query, RowList)] = {
+                if (requestModel.bestCandidates.isDefined
+                  && requestModel.bestCandidates.get.facts.values.map(_.fact.engine).toSet.size > 1
+                  && requestModel.forceQueryEngine.isEmpty) {
+                  val (newFactBestCandidateOption, newBestDimCandidates) = findBestCandidates(requestModel, Set(factBestCandidateOption.get.fact.engine))
```

Comment:
If there is no other fallback engine, this will fail, you probably don't want to have a fallback in that case.

Reply:
that condition exists 
`requestModel.bestCandidates.get.facts.values.map(_.fact.engine).toSet.size > 1`